platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/storage/FileSource.java

```diff
@@ -79,17 +110,47 @@ public static synchronized FileSource getInstance(@NonNull Context context) {
    */
   @NonNull
   private static String getCachePath(@NonNull Context context) {
+    String cachePath = null;
+    if (isExternalStorageConfiguration(context) && isExternalStorageReadable()) {
```

Comment:
We shouldn't worry about these checks at this point, the idea here should be that the path can be completely custom, as long as it's writeable. This check should be moved down to when we failed to obtain a path that is saved in the shared preferences (or it's not writeable) and then check whether the manifest configuration is enabled and use `context.getExternalFilesDir(null).getAbsolutePath()` (or fallback to internal) in order to maintain backward compatibility and don't break semver.

Reply:
Done.